src/callbacks_stage/positivity_zhang_shu_dg1d.jl

```diff
@@ -43,5 +43,63 @@ function limiter_zhang_shu!(u, threshold::Real, variable,
   return nothing
 end
 
+# Overload limiter for ShallowWaterEquations1D and cut off waterheight as well as velocity
+#   to avoid numerical problems/instabilities caused by velocities close to zero
+function limiter_zhang_shu!(u, threshold::Real, variable,
+                            mesh::AbstractMesh{1}, equations::ShallowWaterEquations1D,
+                            dg::DGSEM, cache)
+  @unpack weights = dg.basis
+
+  @threaded for element in eachelement(dg, cache)
+    # determine minimum value
+    value_min = typemax(eltype(u))
+    for i in eachnode(dg)
+      u_node = get_node_vars(u, equations, dg, i, element)
+      value_min = min(value_min, variable(u_node, equations))
+    end
+
+    # detect if limiting is necessary
+    value_min < threshold || continue
+
+    # compute mean value
+    u_mean = zero(get_node_vars(u, equations, dg, 1, element))
+    for i in eachnode(dg)
+      u_node = get_node_vars(u, equations, dg, i, element)
+      u_mean += u_node * weights[i]
+    end
+    # note that the reference element is [-1,1]^ndims(dg), thus the weights sum to 2
+    u_mean = u_mean / 2^ndims(mesh)
+
+    # We compute the value directly with the mean values, as we assume that
+    # Jensen's inequality holds (e.g. pressure for compressible Euler equations).
+    value_mean = variable(u_mean, equations)
+    theta = (value_mean - threshold) / (value_mean - value_min)
+    for i in eachnode(dg)
+      u_node = get_node_vars(u, equations, dg, i, element)
+
+      # Cut velocity off in case of waterheight beeing smaller than threshold
+
+      h_node, h_v_node, b_node = u_node
+      h_mean, h_v_mean, _ = u_mean #b_mean is not used as b_node must not be overwritten
+
+      # Set them both to zero to apply linear combination correctly
+      h_v_node = h_v_node * Int32(h_node > threshold)
+
+      h_v_mean = h_v_mean * Int32(h_node > threshold)
+
+      u_node = SVector(h_node, h_v_node, b_node)
+      u_mean = SVector(h_mean, h_v_mean, 0)
+
+      # When velocity is cut off, the only value beeing averaged is the waterheight,
+      #   because the velocity is set to zero and this value is passed.
+      # Otherwise, the velocity is averaged, as well.
```

Comment:
```suggestion
      # When velocity is cut off, the only averaged value is the waterheight,
      # because the velocity is set to zero.
      # Otherwise, the velocity is averaged, as well.
      # Note that the auxiliary bottom topography variable `b` is never limited.
```

Reply:
All these comments have resolved now that you have implemented a new `PositivityPreservingLimiterShallowWater`